test/optimizer/joins/delim_join_dont_explode.test_slow

```diff
@@ -62,7 +63,7 @@ exists(
 )
 order by bt.id
 ----
-physical_plan	<REGEX>:.*HASH_JOIN.*SEQ_SCAN.*DELIM_SCAN.*SEQ_SCAN.*
+physical_plan	<REGEX>:.*HASH_JOIN.*DELIM_SCAN.*SEQ_SCAN.*SEQ_SCAN.*
```

Comment:
Why has this plan changed?

This the the plan on `main`:
```
┌───────────────────────────┐
│         PROJECTION        │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│        Projections:       │
│__internal_decompress_integ│
│     ral_integer(#0, 0)    │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│   Estimated Cardinality:  │
│            1000           │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│          ORDER_BY         │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│         Order By:         │
│         bt.id ASC         │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│         PROJECTION        │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│        Projections:       │
│__internal_compress_integra│
│     l_usmallint(#0, 0)    │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│   Estimated Cardinality:  │
│            1000           │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│      RIGHT_DELIM_JOIN     │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│         Join Type:        │
│         RIGHT_SEMI        │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│        Conditions:        │
│ id IS NOT DISTINCT FROM id├──────────────┐
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
│   Estimated Cardinality:  │              │
│            1000           │              │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
│        Delim Index:       │              │
│             1             │              │
└─────────────┬─────────────┘              │
┌─────────────┴─────────────┐┌─────────────┴─────────────┐
│         SEQ_SCAN          ││         HASH_JOIN         │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│        Stringified:       ││         Join Type:        │
│         big_table         ││         RIGHT_SEMI        │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│        Projections:       ││        Conditions:        ├────────────────────────────────────────────────────────────────────────┐
│             id            ││ id IS NOT DISTINCT FROM id│                                                                        │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │                                                                        │
│   Estimated Cardinality:  ││   Estimated Cardinality:  │                                                                        │
│            1000           ││            1000           │                                                                        │
└───────────────────────────┘└─────────────┬─────────────┘                                                                        │
                             ┌─────────────┴─────────────┐                                                          ┌─────────────┴─────────────┐
                             │         PROJECTION        │                                                          │         DUMMY_SCAN        │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │                                                          │                           │
                             │        Projections:       │                                                          │                           │
                             │             id            │                                                          │                           │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │                                                          │                           │
                             │   Estimated Cardinality:  │                                                          │                           │
                             │             50            │                                                          │                           │
                             └─────────────┬─────────────┘                                                          └───────────────────────────┘
                             ┌─────────────┴─────────────┐
                             │         HASH_JOIN         │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │         Join Type:        │
                             │           INNER           │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │        Conditions:        │
                             │    id = fk_to_medium_2    │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │         Build Min:        ├──────────────┐
                             │             99            │              │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
                             │         Build Max:        │              │
                             │             99            │              │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
                             │   Estimated Cardinality:  │              │
                             │             50            │              │
                             └─────────────┬─────────────┘              │
                             ┌─────────────┴─────────────┐┌─────────────┴─────────────┐
                             │         SEQ_SCAN          ││         HASH_JOIN         │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │        Stringified:       ││         Join Type:        │
                             │          medium_2         ││           INNER           │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │        Projections:       ││        Conditions:        │
                             │             id            ││ (id % 7) = (fk_to_big % 7)├──────────────┐
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
                             │          Filters:         ││   Estimated Cardinality:  │              │
                             │ id>=99 AND id IS NOT NULL ││            500            │              │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││                           │              │
                             │   Estimated Cardinality:  ││                           │              │
                             │            100            ││                           │              │
                             └───────────────────────────┘└─────────────┬─────────────┘              │
                                                          ┌─────────────┴─────────────┐┌─────────────┴─────────────┐
                                                          │         DELIM_SCAN        ││         SEQ_SCAN          │
                                                          │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                                                          │        Delim Index:       ││        Stringified:       │
                                                          │             1             ││          medium_1         │
                                                          │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                                                          │   Estimated Cardinality:  ││        Projections:       │
                                                          │            500            ││       fk_to_medium_2      │
                                                          │                           ││         fk_to_big         │
                                                          │                           ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                                                          │                           ││          Filters:         │
                                                          │                           ││   fk_to_medium_2<=99 AND  │
                                                          │                           ││ fk_to_medium_2 IS NOT NULL│
                                                          │                           ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                                                          │                           ││   Estimated Cardinality:  │
                                                          │                           ││            100            │
                                                          └───────────────────────────┘└───────────────────────────┘
```
The REGEX tests specifically this part of the query:
```
┌─────────────┴─────────────┐
│         HASH_JOIN         │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│         Join Type:        │
│           INNER           │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│        Conditions:        │
│    id = fk_to_medium_2    │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│         Build Min:        ├──────────────┐
│             99            │              │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
│         Build Max:        │              │
│             99            │              │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
│   Estimated Cardinality:  │              │
│             50            │              │
└─────────────┬─────────────┘              │
┌─────────────┴─────────────┐┌─────────────┴─────────────┐
│         SEQ_SCAN          ││         HASH_JOIN         │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│        Stringified:       ││         Join Type:        │
│          medium_2         ││           INNER           │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
│        Projections:       ││        Conditions:        │
│             id            ││ (id % 7) = (fk_to_big % 7)├──────────────┐
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │              │
│          Filters:         ││   Estimated Cardinality:  │              │
│ id>=99 AND id IS NOT NULL ││            500            │              │
│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││                           │              │
│   Estimated Cardinality:  ││                           │              │
│            100            ││                           │              │
└───────────────────────────┘└─────────────┬─────────────┘              │
                             ┌─────────────┴─────────────┐┌─────────────┴─────────────┐
                             │         DELIM_SCAN        ││         SEQ_SCAN          │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │        Delim Index:       ││        Stringified:       │
                             │             1             ││          medium_1         │
                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │   Estimated Cardinality:  ││        Projections:       │
                             │            500            ││       fk_to_medium_2      │
                             │                           ││         fk_to_big         │
                             │                           ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │                           ││          Filters:         │
                             │                           ││   fk_to_medium_2<=99 AND  │
                             │                           ││ fk_to_medium_2 IS NOT NULL│
                             │                           ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
                             │                           ││   Estimated Cardinality:  │
                             │                           ││            100            │
                             └───────────────────────────┘└───────────────────────────┘
```
Which is just 2 inner joins. Shouldn't this part of the plan stay the same?

Reply:
We discussed this and want to merge the PR as is since the join order does not affect correctness, while the semi/anti join ordering fix does affect correctness. This should be investigated in a future PR.